packages/azure-http-specs/specs/azure/resource-manager/method-subscription-id/main.tsp

```diff
@@ -0,0 +1,201 @@
+  model SubscriptionResource1 is ProxyResource<SubscriptionResource1Properties> {
+    ...ResourceNameParameter<SubscriptionResource1>;
+  }
+
+  @doc("Properties of subscription resource 1.")
+  model SubscriptionResource1Properties {
+    @visibility(Lifecycle.Read)
+    @doc("The status of the last operation.")
+    provisioningState?: ResourceProvisioningState;
+
+    @doc("The description of the resource.")
+    description?: string;
+  }
+
+  @subscriptionResource
+  @resource("subscriptionResource2s")
+  model SubscriptionResource2 is ProxyResource<SubscriptionResource2Properties> {
+    ...ResourceNameParameter<SubscriptionResource2>;
+  }
+
+  @doc("Properties of subscription resource 2.")
+  model SubscriptionResource2Properties {
+    @visibility(Lifecycle.Read)
+    @doc("The status of the last operation.")
+    provisioningState?: ResourceProvisioningState;
+
+    @doc("The configuration value.")
+    configValue?: string;
+  }
+
+  // Define base parameter models to enable subscriptionId parameter access
+  model GetSubscriptionResource1BaseParameter
+    is Azure.ResourceManager.Foundations.DefaultBaseParameters<SubscriptionResource1>;
+  model PutSubscriptionResource1BaseParameter
+    is Azure.ResourceManager.Foundations.DefaultBaseParameters<SubscriptionResource1>;
+  model DeleteSubscriptionResource1BaseParameter
+    is Azure.ResourceManager.Foundations.DefaultBaseParameters<SubscriptionResource1>;
+
+  model GetSubscriptionResource2BaseParameter
+    is Azure.ResourceManager.Foundations.DefaultBaseParameters<SubscriptionResource2>;
+  model PutSubscriptionResource2BaseParameter
+    is Azure.ResourceManager.Foundations.DefaultBaseParameters<SubscriptionResource2>;
+  model DeleteSubscriptionResource2BaseParameter
+    is Azure.ResourceManager.Foundations.DefaultBaseParameters<SubscriptionResource2>;
+
+  @armResourceOperations
+  interface SubscriptionResource1Operations {
+    get is ArmResourceRead<
+      SubscriptionResource1,
+      BaseParameters = GetSubscriptionResource1BaseParameter
+    >;
+    put is ArmResourceCreateOrReplaceAsync<
+      SubscriptionResource1,
+      BaseParameters = PutSubscriptionResource1BaseParameter
+    >;
+    delete is ArmResourceDeleteWithoutOkAsync<
+      SubscriptionResource1,
+      BaseParameters = DeleteSubscriptionResource1BaseParameter
+    >;
+    list is ArmListBySubscription<SubscriptionResource1>;
+  }
+
+  @armResourceOperations
+  interface SubscriptionResource2Operations {
+    get is ArmResourceRead<
+      SubscriptionResource2,
+      BaseParameters = GetSubscriptionResource2BaseParameter
+    >;
+    put is ArmResourceCreateOrReplaceAsync<
+      SubscriptionResource2,
+      BaseParameters = PutSubscriptionResource2BaseParameter
+    >;
+    delete is ArmResourceDeleteWithoutOkAsync<
+      SubscriptionResource2,
+      BaseParameters = DeleteSubscriptionResource2BaseParameter
+    >;
+    list is ArmListBySubscription<SubscriptionResource2>;
```

Comment:
it seems we only need two get operations in different interfaces? no need for other put or delete?

Reply:
added.